python/requirements-wheel-test.txt

```diff
@@ -8,14 +8,18 @@ pytz
 tzdata; sys_platform == 'win32'
 
 numpy==1.19.5; platform_system == "Linux"   and platform_machine == "aarch64" and python_version <  "3.7"
-numpy==1.21.3; platform_system == "Linux"   and platform_machine == "aarch64" and python_version >= "3.7"
+numpy==1.21.3; platform_system == "Linux"   and platform_machine == "aarch64" and python_version >= "3.7" and python_version < "3.11"
+numpy==1.23.4; platform_system == "Linux"   and platform_machine == "aarch64" and python_version >= "3.11"
 numpy==1.19.5; platform_system == "Linux"   and platform_machine != "aarch64" and python_version <  "3.9"
-numpy==1.21.3; platform_system == "Linux"   and platform_machine != "aarch64" and python_version >= "3.9"
+numpy==1.21.3; platform_system == "Linux"   and platform_machine != "aarch64" and python_version >= "3.9" and python_version < "3.11"
+numpy==1.23.4; platform_system == "Linux"   and platform_machine != "aarch64" and python_version >= "3.11"
 numpy==1.21.3; platform_system == "Darwin"  and platform_machine == "arm64"
 numpy==1.19.5; platform_system == "Darwin"  and platform_machine != "arm64"   and python_version <  "3.9"
-numpy==1.21.3; platform_system == "Darwin"  and platform_machine != "arm64"   and python_version >= "3.9"
+numpy==1.21.3; platform_system == "Darwin"  and platform_machine != "arm64"   and python_version >= "3.9" and python_version < "3.11"
+numpy==1.23.4; platform_system == "Darwin"  and platform_machine != "arm64"   and python_version >= "3.11"
 numpy==1.19.5; platform_system == "Windows"                                   and python_version <  "3.9"
-numpy==1.21.3; platform_system == "Windows"                                   and python_version >= "3.9"
+numpy==1.21.3; platform_system == "Windows"                                   and python_version >= "3.9" and python_version < "3.11"
+numpy==1.23.4; platform_system == "Windows"                                   and python_version >= "3.11"
```

Comment:
Is this copied from `oldest-supported-numpy`, or did we do our own thing?

Reply:
you mean the origin of this file? I am not sure about the origin of it but it currently fails due to trying to install `numpy==1.21.3` for Python 3.11 on our test wheels container when the first numpy version that support Python 3.11 is 1.23.3 (https://pypi.org/project/numpy/1.23.3/)

Reply:
This file is for testing, not building, I think? (so it doesn't need to match exactly with the versions from oldest-supported-numpy)

Reply:
I have created a follow up ticket (https://issues.apache.org/jira/browse/ARROW-18170) to see whether we can get rid of this by using the `oldest-supported-numpy` as the current requirements-wheel-test.txt feels overly complex.